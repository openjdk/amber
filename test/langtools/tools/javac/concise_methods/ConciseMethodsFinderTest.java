/*
 * @test /nodynamiccopyright/
 * @summary smoke test for the concise method finder
 * @compile/fail/ref=ConciseMethodsFinderTest.out -Werror -XDrawDiagnostics -XDfind=concise_method ConciseMethodsFinderTest.java
 */

class ConciseMethodsFinderTest {

    class Inner {
        String si;
    }

    Inner inner;

    Inner getInner() { return inner; }

    ConciseMethodsFinderTest(String s, String p) {}

    // this one is a candidate
    int length(String s) {
        return s.length();
    }

    // this one is a candidate
    int length2() {
        return inner.si.length();
    }

    // this one is a candidate
    int length3() {
        return getInner().si.length();
    }

    // this one is a candidate
    static ConciseMethodsFinderTest makeTest1(String s, String k) {
        return new ConciseMethodsFinderTest(s, k);
    }

    // not a candidate
    static ConciseMethodsFinderTest makeTest2(String s) {
        return new ConciseMethodsFinderTest(s, "");
    }

    // not a candidate
    int length2(String s, boolean b) { return s.length(); }

    // this one is a candidate
    String test() {
        return new Object(){}.toString();
    }
}
