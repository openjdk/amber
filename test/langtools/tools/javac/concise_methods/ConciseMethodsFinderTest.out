ConciseMethodsFinderTest.java:20:9: compiler.warn.potential.concise.method.found
ConciseMethodsFinderTest.java:25:9: compiler.warn.potential.concise.method.found
ConciseMethodsFinderTest.java:30:9: compiler.warn.potential.concise.method.found
ConciseMethodsFinderTest.java:35:37: compiler.warn.potential.concise.method.found
ConciseMethodsFinderTest.java:48:12: compiler.warn.potential.concise.method.found
- compiler.err.warnings.and.werror
1 error
5 warnings
